test/integration/orderFlowFee.behavior.t.sol

```diff
@@ -107,57 +115,275 @@ contract OrderFlowFeeTest is Test, ConsolidatedEvents {
+    function test_imposeOrderFlowFee_market_margin_failed() public {
+        // Set orderflow fee high to easily test that transaction fails if neither account or market has sufficient margin to cover for fees
+        uint256 testOrderFlowFee = 10_000; // 10%
+        settings.setOrderFlowFee(testOrderFlowFee);
+
+        fundAccount(1000 ether);
+
+        uint256 treasuryPreBalance = sUSD.balanceOf(settings.TREASURY());
+
+        // create a long position in the ETH market
+        address market = getMarketAddressFromKey(sETHPERP);
+
+        /// Deposit all available account margin into market margin
+        int256 marginDelta = int256(1000 ether);
+        int256 sizeDelta = 1 ether;
+
+        (uint256 desiredFillPrice,) =
+            IPerpsV2MarketConsolidated(market).assetPrice();
+
+        // Deposit market margin
+        IAccount.Command[] memory commands = new IAccount.Command[](1);
+        commands[0] = IAccount.Command.PERPS_V2_MODIFY_MARGIN;
+        bytes[] memory inputs = new bytes[](1);
+        inputs[0] = abi.encode(market, marginDelta);
+        account.execute(commands, inputs);
+
+        // Execute Atomic Order
+        IAccount.Command[] memory commandsAtomic = new IAccount.Command[](2);
+        commandsAtomic[0] = IAccount.Command.PERPS_V2_SUBMIT_ATOMIC_ORDER;
+        bytes[] memory inputsAtomic = new bytes[](2);
+        inputsAtomic[0] = abi.encode(market, sizeDelta, desiredFillPrice);
+
+        // Current configuration should have insufficient margin to cover for orderflow fee and revert
+        // because there is not enough margin for the position delta.
+        vm.expectRevert("Insufficient margin");
 
-    /// @custom:todo use atomic order flow to simplify testing
+        account.execute(commandsAtomic, inputsAtomic);
+
+        uint256 treasuryPostBalance = sUSD.balanceOf(settings.TREASURY());
+
+        // Assert that no overflowfee was distributed
+        assertEq(treasuryPreBalance, treasuryPostBalance);
+    }
+
+    /// Verifies that the correct Event is emitted with correct fee value
     function test_imposeOrderFlowFee_event() public {
-        /// @custom:todo test the following:
-        /// 1. is the correct event emitted
-        /// 2. is the correct fee emitted with it
+        fundAccount(AMOUNT);
+        // create a long position in the ETH market
+        address market = getMarketAddressFromKey(sETHPERP);
+
+        /// Keep account margin to cover for orderFlowFee
+        int256 marginDelta = int256(AMOUNT) / 5;
+        int256 sizeDelta = 1;
+
+        (uint256 desiredFillPrice,) =
+            IPerpsV2MarketConsolidated(market).assetPrice();
+
+        vm.expectEmit(true, true, true, true);
+        // orderFlowFee is 94025250000000000 in this configuration
+        emit OrderFlowFeeImposed(address(account), 94_025_250_000_000_000);
+
+        submitAtomicOrder(sETHPERP, marginDelta, sizeDelta, desiredFillPrice);
+    }
+
```

Comment:
would be good to have these tests where there is a pending conditional order and thus the free margin != actual margin in the account. This is just to make sure "locked" account margin is never used to pay the fee.

Reply:
Added test_imposeOrderFlowFee_market_margin_with_pending_order() for this